source/geometries/NextDemoSiPMBoard.cc

```diff
@@ -208,25 +211,59 @@ void NextDemoSiPMBoard::Construct()
 
 
   /// Mask Holes
+  G4String coating_name = "BOARD_COATING";
+  G4Material* tpb = MaterialsList::TPB();
+  tpb->SetMaterialPropertiesTable(OpticalMaterialProperties::TPB());
+  G4OpticalSurface* coating_opsurf =
+    new G4OpticalSurface(coating_name + "_OPSURF", glisur, ground,
+                         dielectric_dielectric, .01);
+
   G4VSolid* hole_solid = nullptr;
   G4String hole_name = "BOARD_MASK_HOLE";
   if (hole_type_ == "rounded"){
     hole_solid = new G4Tubs(hole_name, 0., hole_diam_/2., mask_thickn_/2., 0, 360.*deg);}
   else if (hole_type_ == "rectangular"){
     hole_solid = new G4Box(hole_name, hole_x_/2., hole_y_/2., mask_thickn_/2.);}
 
-  G4LogicalVolume* hole_logic =
-    new G4LogicalVolume(hole_solid, mother_gas, hole_name);
+  G4LogicalVolume* hole_logic = new G4LogicalVolume(hole_solid, mother_gas, hole_name);
 
   // Generate SiPM positions
   GenerateSiPMPositions();
 
   G4double sipm_posz = - mask_thickn_/2. + sipm_z_dim/2.;
 
-  new G4PVPlacement(sipm_rot, G4ThreeVector(0., 0., sipm_posz), sipm_->GetLogicalVolume(),
-                    sipm_->GetLogicalVolume()->GetName(), hole_logic,
-                    false, 0, false);
+  G4VPhysicalVolume* hole_phys_ = new G4PVPlacement(sipm_rot, G4ThreeVector(0., 0., sipm_posz), sipm_->GetLogicalVolume(),
+                                                    sipm_->GetLogicalVolume()->GetName(), hole_logic,
+                                                    false, 0, false);
+
+  if (hole_coated_ && (hole_type_ == "rectangular")){
+
+    G4Box* hole_coating_top = new G4Box("HOLE_COATING", hole_x_/2., coating_thickn_/2., mask_thickn_/2.);
+    G4Box* hole_coating_lat = new G4Box("HOLE_COATING", coating_thickn_/2., hole_y_/2., mask_thickn_/2.);
+
+    G4UnionSolid* hole_coating = new G4UnionSolid("HOLE_COATING", hole_coating_top, hole_coating_lat, 0,
+                                                  G4ThreeVector(hole_x_/2.-coating_thickn_/2., -hole_y_/2. + coating_thickn_/2., 0.));
 
+    hole_coating = new G4UnionSolid("HOLE_COATING", hole_coating, hole_coating_lat, 0,
+                                    G4ThreeVector(-hole_x_/2.+coating_thickn_/2., -hole_y_/2. + coating_thickn_/2., 0.));
+
+    hole_coating = new G4UnionSolid("HOLE_COATING", hole_coating, hole_coating_top, 0,
+                                    G4ThreeVector(0., -hole_y_ + coating_thickn_, 0.));
+
+    G4LogicalVolume* hole_coating_logic = new G4LogicalVolume(hole_coating, tpb, "HOLE_COATING");
+
+    G4VPhysicalVolume* hole_coating_phys_ =
+          new G4PVPlacement(nullptr, G4ThreeVector(0., hole_y_/2. - coating_thickn_/2., 0.), hole_coating_logic,
+                            "HOLE_COATING", hole_logic, false, 0, false);
+
+    new G4LogicalBorderSurface("TEFLON_WLS_HOLE_GAS_OPSURF", hole_coating_phys_,
```

Comment:
If you use `hole_phys_` here, you're setting the surface between the SiPM and the coating, not between the gas of the hole and the coating, because `hole_phys_` is the physical volume of the SiPM in the current version of the code, right?

Reply:
The sipm volume is smaller than the hole volume. The hole has dimensions of the mask holes and sipms are placed inside. That means that most of the hole is filled with Xe, except the SiPM volume, in the  figure the solid box represents the full SiPM encasing and the coated walls:
<img width="369" alt="SiPM-inside-holes" src="https://user-images.githubusercontent.com/35993868/123936595-9e0def80-d995-11eb-9914-75a1d5896ca2.png">

Reply:
Right, `hole_phys` should be `sipm_phys`. These `G4LogicalBorderSurface` should be defined for each hole in its positioning at line 289, no?

Reply:
I would say so. However, I'm looking at our own documentation of the optical surfaces (https://github.com/next-exp/nexus/wiki/Setting-optical-properties) and there's a caveat regarding the case when one of the volumes is the daughter of the other one. I think we should try and find a time to talk to @jmunozv to clarify this.